lib/Str.js

```diff
@@ -0,0 +1,30 @@
+import _ from 'underscore';
+
+const Str = {
+    /**
+     * Returns true if the haystack begins with the needle
+     *
+     * @param {String} haystack  The full string to be searched
+     * @param {String} needle    The case-sensitive string to search for
+     * @return {Boolean} Retruns true if the haystack starts with the needle.
+     */
+    startsWith(haystack, needle) {
+        return _.isString(haystack)
+                && _.isString(needle)
+                && haystack.substring(0, needle.length) === needle;
+    },
+
+    /**
+     * Checks if parameter is a string or function
+     * if it is a function then we will call it with
+     * any additional arguments.
+     *
+     * @param {String|Function} parameter
+     * @returns {String}
```

Comment:
This doesn't necessarily return a string, can we update it?

Reply:
I think if a function is passed to the first argument (instead of string), that function should return a string.

I don't know how to provide type information of callbacks, but hopefully when we migrate onyx, we can type that.

Reply:
This function is pretty weird 🙃

Reply:
This function returns `string` if we pass a `string` to `parameter`, or returns the result of `parameter` if it's a function. So, I will change to `{*}` as if we pass a function it can return any type. WDYT?

Reply:
@hayata-suenaga In this [example](https://github.com/Expensify/react-native-onyx/blob/main/lib/withOnyx.js#L121) the function is returning a `boolean` instead of a `string`.

Reply:
I just don't know why this function exists 😭